sites/friday/src/get-started/nextjs/add-database.md

```diff
@@ -0,0 +1,228 @@
+  | mariadb          | 0.5  | 0.5 | 1408        | 512       | 1         |
+  +------------------+------+-----+-------------+-----------+-----------+
+
+Summary of changes:
+  Service: mariadb
+    CPU: 0.5
+    Memory: 1408 MB
+    Disk: 512 MB
+
+Are you sure you want to continue? [Y/n] y
+```
+
+## 4. Configure your Express application to use this new database
+First, you need a NodeJs module named ``mysql2``. Install it by running the following command:
+```bash {location="Terminal"}
+npm install mariadb
+```
+
+Then modify your index.js with the following :
+```javascript {location="index.js"}
+const express = require('express')
+const app = express()
+const mysql = require("mysql2/promise");
+const port = (process.env.PORT || '3000');
+
+function openConnection() {
+  return mysql.createConnection({
+    host: process.env.DB_HOST,
+    port: process.env.DB_PORT,
+    user: process.env.DB_USERNAME,
+    password: process.env.DB_PASSWORD,
+    database: process.env.DB_DATABASE
+  });
+}
+
+function createTable(connection) {
+  return connection.execute(
+    `CREATE TABLE IF NOT EXISTS {{% vendor/cli %}}info (
+      uid INT(10) NOT NULL AUTO_INCREMENT,
+      username VARCHAR(64) NULL DEFAULT NULL,
+      departname VARCHAR(128) NULL DEFAULT NULL,
+      created DATE NULL DEFAULT NULL,
+      PRIMARY KEY (uid)
+    ) DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;`
+  );
+}
+
+function insertData(connection) {
+  return connection.execute(
+    "INSERT INTO {{% vendor/cli %}}info (username, departname, created) VALUES ('{{% vendor/cli %}}', 'Deploy Friday', '2023-09-29')"
+  );
+}
+
+function readData(connection) {
+  return connection.query("SELECT * FROM {{% vendor/cli %}}info");
+}
+
+function dropTable(connection) {
+  return connection.execute("DROP TABLE {{% vendor/cli %}}info");
+}
+
+// Define the main route.
+app.get('/', async function(req, res){
+
+  // Connect to MariaDB.
+  const connection = await openConnection();
+
+  await createTable(connection);
+  await insertData(connection);
+
+  const [rows] = await readData(connection);
+
+  const droppedResult = await dropTable(connection);
+
+  // Make the output.
+  const outputString = `Hello, World! - A simple Express web framework template for {{% vendor/name %}}
+
+MariaDB Tests:
+
+* Connect and add row:
+  - Row ID (1): ${rows[0].uid}
+  - Username ({{% vendor/cli %}}): ${rows[0].username}
+  - Department (Deploy Friday): ${rows[0].departname}
+  - Created (2023-09-29): ${rows[0].created}
+* Delete row:
+  - Status (0): ${droppedResult[0].warningStatus}`;
+
+  res.set('Content-Type', 'text/plain');
+  res.send(outputString);
+});
+
+// Get PORT and start the server
+app.listen(port, function() {
+  console.log(`Listening on port ${port}`)
+});
+```
+
+Commit your changes and deploy your changes:
+```bash {location="Terminal"}
+git add package.json package-lock.json index.js && git commit -m "adding MariaDb database service"
+{{% vendor/cli %}} push
```

Comment:
I'm failing at this point with an error of 
```
Module not found: Can't resolve 'express'
```
But maybe this is where you stopped working yesterday?

Reply:
same